### PR TITLE
Add roles CLI

### LINKED ISSUE #8744: RBAC: CLI interface
### Description and goal

Refs #8171

Implement a CLI interface to ease role management.

CLI:

- [ ] `editoast roles add SUBJECT ROLE [ROLE+]`
- [ ] `editoast roles remove SUBJECT ROLE [ROLE+]`
- [ ] `editoast roles list SUBJECT`
- [ ] `editoast roles list-roles`

where:
* `SUBJECT` can be either a subject id (u64) or a user identity as provided by the gateway (String)
* `ROLE` is a builtin role identifier
* `list-roles` lists the builtin roles tag

### Acceptance criteria

The commands are implemented.

> [!NOTE]
> There will be a CLI to manage users and groups as well. Subject IDs will be obtainable from there.

### PR DESCRIPTION
**Pull request by leovalais**
closes #8744

### CODE REVIEW

File: editoast/src/client/roles.rs [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+}
+
+pub fn list_roles() {
+    BuiltinRole::iter().for_each(|role| println!("{role}"));
+}
+
+async fn parse_and_fetch_subject_id(
+    subject: &String,
+    driver: &PgAuthDriver<BuiltinRole>,
+) -> anyhow::Result<i64> {
```
Comment (flomonster) [inline editoast/src/client/roles.rs]: We could return more information to have more user friendly prompt:

```diff
$ editoast roles list 2
- Subject 2 has no roles assigned
+ Subject flomonster (2) has no roles assigned
```

```suggestion
) -> anyhow::Result<UserInfo> {
```

File: editoast/src/client/roles.rs [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+pub async fn add_roles(
+    AddArgs { subject, roles }: AddArgs,
+    conn: DbConnection,
+) -> anyhow::Result<()> {
+    let driver = PgAuthDriver::<BuiltinRole>::new(conn);
+    let subject_id = parse_and_fetch_subject_id(&subject, &driver).await?;
+    let roles = roles
+        .iter()
+        .map(|role| {
+            role.parse::<BuiltinRole>()
```
Comment (flomonster) [inline editoast/src/client/roles.rs]: This auto implementation is case sensitive which is not ideal. We could:
- Implement manually `std::str::FromStr` to be case insensitive.
- Add a lowercase version of all variant and lower case the input role.

**Review by flomonster — APPROVED** [review]
LGTM.

I came across two irritating little things while testing the CLI.

File: editoast/src/client/roles.rs (hunk 2 above)
Comment (leovalais) [reply editoast/src/client/roles.rs]: I'm a bit hesitant to add such DoF to the builtin roles serialization at the API level just for this use case.

I've implemented this behavior in the parsing step directly (admittedly it's a O(n^2) operation but it's in the CLI honestly who cares). I'll let you resolve if it's ok.

**Review by woshilapin — CHANGES_REQUESTED** [review]
Tested and might be missing some information on the CLI. Here is an example (I removed all `tracing` logs).

```
❯ export RUST_LOG=none 
❯ editoast roles list 1
morane#1 (Bob Morane) has no roles assigned

❯ editoast roles add 1 SuperUser StdcmAdmin

❯ editoast roles list 1
Superuser
StdcmAdmin

❯ editoast roles remove 1 SuperUser StdcmAdmin

```

A few things to note that could be improved.
- When adding roles, there is no message confirming everything went fine.
- When removing roles, there is no message confirming everything went fine.
- When adding or removing roles of a user that doesn't exist, there is also no message.
- When listing roles and the user has no roles, the user and its ID are clearly identified... but not when the user has roles, only the roles are displayed: maybe having also the user printed would be nice

Another problem I see is that I would expect `add` to append roles, but it actually overwrites the existing list. See for example.

```
❯ editoast roles add 1 SuperUser StdcmAdmin
❯ editoast roles list 1
Superuser
StdcmAdmin
❯ editoast roles add 1 OpsWrite
❯ editoast roles list 1
OpsWrite
```
I would expect the last command to output 3 values.

Would it make sense to add some tests about it?

**Comment by leovalais** [comment]
As discussed:

> Another problem I see is that I would expect add to append roles, but it actually overwrites the existing list. 

Can't reproduce.

> A few things to note that could be improved.
> When adding roles, there is no message confirming everything went fine.
>When removing roles, there is no message confirming everything went fine.
>When adding or removing roles of a user that doesn't exist, there is also no message.
>When listing roles and the user has no roles, the user and its ID are clearly identified... but not when the user has roles, only the roles are displayed: maybe having also the user printed would be nice

My logic for choosing between logging and `println!` was that logging is only meant to be informative and provide context while `println!` is an output that can be used in a script (for piping into another process for ex.). This is debatable and can be changed ofc.

@woshilapin Wdyt?

**Comment by woshilapin** [comment]
> As discussed:
> 
> > Another problem I see is that I would expect add to append roles, but it actually overwrites the existing list.
> 
> Can't reproduce.

Can't reproduce either, I might have messed up somewhere, sorry.

> My logic for choosing between logging and `println!` was that logging is only meant to be informative and provide context while `println!` is an output that can be used in a script (for piping into another process for ex.). This is debatable and can be changed ofc.
> 
> @woshilapin Wdyt?

I agree, that was a point of view I didn't have and a totally valid one. Fine with me.

**Review by woshilapin — APPROVED** [review]
Thank you. Let's merge that 😄

**Review by flomonster — APPROVED** [review]
Thanks!

**Pull request merged**